internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -179,6 +189,105 @@ func getOwnerReferenceFrom(obj, owner client.Object) *metav1.OwnerReference {
 	return nil
 }
 
+func (r *Reconciler) callAfterHooks(ctx context.Context, s *scope.Scope) error {
+	if err := r.callAfterControlPlaneInitialized(ctx, s); err != nil {
+		return err
+	}
+
+	if err := r.callAfterClusterUpgrade(ctx, s); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) callAfterControlPlaneInitialized(ctx context.Context, s *scope.Scope) error {
+	/*
+		TODO: Working comment - DELETE AFTER:
+		- If the cluster topology is being created then mark the AfterControlPlaneInitialized hook so that we can call it later.
+	*/
+	if s.Current.Cluster.Spec.InfrastructureRef == nil && s.Current.Cluster.Spec.ControlPlaneRef == nil {
+		if err := hooks.Mark(ctx, runtimehooksv1.AfterControlPlaneInitialized, s.Current.Cluster, r.Client); err != nil {
+			return errors.Wrapf(err, "failed to mark %s hook", runtimecatalog.HookName(runtimehooksv1.AfterControlPlaneInitialized))
+		}
+	}
+
+	if hooks.IsMarked(runtimehooksv1.AfterControlPlaneInitialized, s.Current.Cluster) {
+		for _, condition := range s.Current.Cluster.GetConditions() {
```

Comment:
please move the condition check logic into a util func, e.g. isControlPlaneInitialized